[Site Isolation] Add initial RemoteMediaSessionManager and RemoteMediaSessionManagerProxy classes

eric-carlson opened the pull request:
#### c4a455b4856e66bfdf81b16e8655018e90eecedf
<pre>
[Site Isolation] Begin filling out and connecting RemoteMediaSessionManager and RemoteMediaSessionManagerProxy
<a href="https://bugs.webkit.org/show_bug.cgi?id=302546">https://bugs.webkit.org/show_bug.cgi?id=302546</a>
<a href="https://rdar.apple.com/164742719">rdar://164742719</a>

Reviewed by NOBODY (OOPS!).

Begin filling in the implementation of RemoteMediaSessionManager and RemoteMediaSessionManagerProxy
and the client proxies.

* Source/WebCore/platform/audio/MediaSessionManagerInterface.cpp:
(WebCore::MediaSessionManagerInterface::addSession):
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::logger const):
* Source/WebCore/platform/audio/PlatformMediaSession.h:
(WebCore::PlatformMediaSession::create): Deleted.
(WebCore::PlatformMediaSession::PlatformMediaSession): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionInterface::setMediaSessionIdentifier):
(WebCore::PlatformMediaSessionClient::displayType const): Deleted.
(WebCore::PlatformMediaSessionClient::resumeAutoplaying): Deleted.
(WebCore::PlatformMediaSessionClient::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSessionClient::isSuspended const): Deleted.
(WebCore::PlatformMediaSessionClient::isPlaying const): Deleted.
(WebCore::PlatformMediaSessionClient::isAudible const): Deleted.
(WebCore::PlatformMediaSessionClient::isEnded const): Deleted.
(WebCore::PlatformMediaSessionClient::mediaSessionDuration const): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSessionClient::wirelessRoutesAvailableDidChange): Deleted.
(WebCore::PlatformMediaSessionClient::setWirelessPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSessionClient::setShouldPlayToPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionClient::playbackTargetPickerWasDismissed): Deleted.
(WebCore::PlatformMediaSessionClient::isPlayingOnSecondScreen const): Deleted.
(WebCore::PlatformMediaSessionClient::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSessionClient::processIsSuspendedChanged): Deleted.
(WebCore::PlatformMediaSessionClient::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSessionClient::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSessionClient::nowPlayingInfo const): Deleted.
(WebCore::PlatformMediaSessionClient::selectBestMediaSession): Deleted.
(WebCore::PlatformMediaSessionClient::protectedLogger const): Deleted.
(WebCore::PlatformMediaSessionInterface::mediaType const): Deleted.
(WebCore::PlatformMediaSessionInterface::presentationType const): Deleted.
(WebCore::PlatformMediaSessionInterface::suspendBuffering): Deleted.
(WebCore::PlatformMediaSessionInterface::resumeBuffering): Deleted.
(WebCore::PlatformMediaSessionInterface::canReceiveRemoteControlCommands const): Deleted.
(WebCore::PlatformMediaSessionInterface::displayType const): Deleted.
(WebCore::PlatformMediaSessionInterface::supportsSeeking const): Deleted.
(WebCore::PlatformMediaSessionInterface::isSuspended const): Deleted.
(WebCore::PlatformMediaSessionInterface::isPlaying const): Deleted.
(WebCore::PlatformMediaSessionInterface::isAudible const): Deleted.
(WebCore::PlatformMediaSessionInterface::isEnded const): Deleted.
(WebCore::PlatformMediaSessionInterface::duration const): Deleted.
(WebCore::PlatformMediaSessionInterface::shouldOverrideBackgroundLoadingRestriction const): Deleted.
(WebCore::PlatformMediaSessionInterface::isPlayingToWirelessPlaybackTarget const): Deleted.
(WebCore::PlatformMediaSessionInterface::setPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionInterface::externalOutputDeviceAvailableDidChange): Deleted.
(WebCore::PlatformMediaSessionInterface::setShouldPlayToPlaybackTarget): Deleted.
(WebCore::PlatformMediaSessionInterface::playbackTargetPickerWasDismissed): Deleted.
(WebCore::PlatformMediaSessionInterface::requiresPlaybackTargetRouteMonitoring const): Deleted.
(WebCore::PlatformMediaSessionInterface::canProduceAudio const): Deleted.
(WebCore::PlatformMediaSessionInterface::hasMediaStreamSource const): Deleted.
(WebCore::PlatformMediaSessionInterface::resetPlaybackSessionState): Deleted.
(WebCore::PlatformMediaSessionInterface::hasPlayedAudiblySinceLastInterruption const): Deleted.
(WebCore::PlatformMediaSessionInterface::setHasPlayedAudiblySinceLastInterruption): Deleted.
(WebCore::PlatformMediaSessionInterface::shouldOverridePauseDuringRouteChange const): Deleted.
(WebCore::PlatformMediaSessionInterface::nowPlayingInfo const): Deleted.
(WebCore::PlatformMediaSessionInterface::isNowPlayingEligible const): Deleted.
(WebCore::PlatformMediaSessionInterface::updateMediaUsageIfChanged): Deleted.
(WebCore::PlatformMediaSessionInterface::isLongEnoughForMainContent const): Deleted.
(WebCore::PlatformMediaSessionInterface::mediaSessionIdentifier const): Deleted.
(WebCore::PlatformMediaSessionInterface::audioSessionCategoryChanged): Deleted.
(WebCore::PlatformMediaSessionInterface::mediaSessionGroupIdentifier const): Deleted.
(WebCore::PlatformMediaSessionInterface::isPlayingOnSecondScreen const): Deleted.
(WebCore::PlatformMediaSessionInterface::invalidateClient): Deleted.
(WebCore::PlatformMediaSessionInterface::client const): Deleted.
(WebCore::PlatformMediaSessionInterface::protectedLogger const): Deleted.
(WebCore::PlatformMediaSessionInterface::PlatformMediaSessionInterface): Deleted.
(WebCore::PlatformMediaSessionInterface::sessionManager const): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSessionManager.cpp:
(WebCore::PlatformMediaSessionManager::sessions const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.h:
* Source/WebCore/platform/audio/cocoa/MediaSessionManagerCocoa.mm:
(WebCore::MediaSessionManagerCocoa::updateSessionState):
(WebCore::MediaSessionManagerCocoa::scheduleSessionStatusUpdate):
(WebCore::MediaSessionManagerCocoa::addSession):
(WebCore::MediaSessionManagerCocoa::removeSession):
(WebCore::MediaSessionManagerCocoa::setCurrentSession):
(WebCore::MediaSessionManagerCocoa::addSupportedCommand):
(WebCore::MediaSessionManagerCocoa::removeSupportedCommand):
(WebCore::MediaSessionManagerCocoa::supportedCommands const):
(WebCore::MediaSessionManagerCocoa::updateNowPlayingInfo):
* Source/WebCore/platform/audio/ios/MediaSessionManagerIOS.h:
* Source/WebKit/Sources.txt:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp:
(WebKit::RemoteMediaSessionClientProxy::RemoteMediaSessionClientProxy):
(WebKit::RemoteMediaSessionClientProxy::~RemoteMediaSessionClientProxy):
(WebKit::RemoteMediaSessionClientProxy::resumeAutoplaying):
(WebKit::RemoteMediaSessionClientProxy::mayResumePlayback):
(WebKit::RemoteMediaSessionClientProxy::suspendPlayback):
(WebKit::RemoteMediaSessionClientProxy::didReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionClientProxy::shouldOverrideBackgroundPlaybackRestriction const):
(WebKit::RemoteMediaSessionClientProxy::setShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionClientProxy::sessionManager const):
(WebKit::RemoteMediaSessionClientProxy): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.h:
(WebKit::RemoteMediaSessionClientProxy::create): Deleted.
(WebKit::RemoteMediaSessionClientProxy::selectBestMediaSession): Deleted.
(WebKit::RemoteMediaSessionClientProxy::protectedLogger const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.cpp:
(WebKit::RemoteMediaSessionManagerProxy::RemoteMediaSessionManagerProxy):
(WebKit::RemoteMediaSessionManagerProxy::addMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::removeMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::setCurrentMediaSession):
(WebKit::RemoteMediaSessionManagerProxy::updateMediaSessionState):
(WebKit::RemoteMediaSessionManagerProxy::findSession):
(WebKit::RemoteMediaSessionManagerProxy::addSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::removeSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::setCurrentSession): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::updateSessionState): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.h:
(WebKit::RemoteMediaSessionManagerProxy::process const):
(WebKit::RemoteMediaSessionManagerProxy::ref const): Deleted.
(WebKit::RemoteMediaSessionManagerProxy::deref const): Deleted.
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.messages.in:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.cpp:
(WebKit::RemoteMediaSessionProxy::RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::~RemoteMediaSessionProxy):
(WebKit::RemoteMediaSessionProxy::updateState):
(WebKit::RemoteMediaSessionProxy::setShouldPlayToPlaybackTarget):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionProxy.h: Added.
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp:
(WebKit::RemoteMediaSessionManager::addSession):
(WebKit::RemoteMediaSessionManager::removeSession):
(WebKit::RemoteMediaSessionManager::setCurrentSession):
(WebKit::RemoteMediaSessionManager::updateSessionState):
(WebKit::RemoteMediaSessionManager::sessionWithIdentifier):
(WebKit::RemoteMediaSessionManager::clientShouldResumeAutoplaying):
(WebKit::RemoteMediaSessionManager::clientMayResumePlayback):
(WebKit::RemoteMediaSessionManager::clientShouldSuspendPlayback):
(WebKit::RemoteMediaSessionManager::clientSetShouldPlayToPlaybackTarget):
(WebKit::RemoteMediaSessionManager::clientDidReceiveRemoteControlCommand):
(WebKit::RemoteMediaSessionManager::currentSessionState):
(WebKit::RemoteMediaSessionManager::updateCachedSessionState):
(WebKit::RemoteMediaSessionManager::fullSessionState):
(WebKit::RemoteMediaSessionManager::sharedPreferencesForWebProcess const):
(WebKit::platformSessionState): Deleted.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.messages.in:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.h:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.serialization.in:
</pre>

aestes commented on Source/WebKit/UIProcess/WebPageProxy.cpp:
```diff
@@ -10004,6 +10008,14 @@ void WebPageProxy::setMockVideoPresentationModeEnabled(bool enabled)
 #endif
 
+#if ENABLE(VIDEO) || ENABLE(WEB_AUDIO)
+void WebPageProxy::createRemoteMediaSessionManagerProxy()
```
Nit: maybe `ensureRemoteMediaSessionManagerProxy` is a better name. I'd expect something starting with `create` to return something.

aestes commented on Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.h:
```diff
@@ -0,0 +1,81 @@
+        uint64_t messageSenderDestinationID() const final;
+
+    private:
+
```
Nit: extra newline

aestes reviewed (approved):
r=me. Please check if the EWS failures are related to this PR before merging.

eric-carlson committed:
[Site Isolation] Add initial RemoteMediaSessionManager and RemoteMediaSessionManagerProxy classes
https://bugs.webkit.org/show_bug.cgi?id=301005
rdar://162890687

Reviewed by Andy Estes.

Add initial, non-functional, RemoteMediaSessionManager and RemoteMediaSessionManagerProxy
classes with some of the supporting classes and structs that they will use.

* Source/WTF/Scripts/Preferences/UnifiedWebPreferences.yaml:
* Source/WebCore/platform/audio/MediaSessionManagerInterface.h:
(WebCore::MediaSessionManagerInterface::setShouldDeactivateAudioSession): Deleted.
(WebCore::MediaSessionManagerInterface::shouldDeactivateAudioSession): Deleted.
(WebCore::MediaSessionManagerInterface::nowPlayingInfo const): Deleted.
(WebCore::MediaSessionManagerInterface::hasActiveNowPlayingSession const): Deleted.
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingTitle const): Deleted.
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingDuration const): Deleted.
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingElapsedTime const): Deleted.
(WebCore::MediaSessionManagerInterface::lastUpdatedNowPlayingInfoUniqueIdentifier const): Deleted.
(WebCore::MediaSessionManagerInterface::registeredAsNowPlayingApplication const): Deleted.
(WebCore::MediaSessionManagerInterface::haveEverRegisteredAsNowPlayingApplication const): Deleted.
(WebCore::MediaSessionManagerInterface::resetHaveEverRegisteredAsNowPlayingApplicationForTesting): Deleted.
(WebCore::MediaSessionManagerInterface::willIgnoreSystemInterruptions const): Deleted.
(WebCore::MediaSessionManagerInterface::setWillIgnoreSystemInterruptions): Deleted.
(WebCore::MediaSessionManagerInterface::sessionDidEndRemoteScrubbing): Deleted.
(WebCore::MediaSessionManagerInterface::clientCharacteristicsChanged): Deleted.
(WebCore::MediaSessionManagerInterface::configureWirelessTargetMonitoring): Deleted.
(WebCore::MediaSessionManagerInterface::hasWirelessTargetsAvailable): Deleted.
(WebCore::MediaSessionManagerInterface::isMonitoringWirelessTargets const): Deleted.
(WebCore::MediaSessionManagerInterface::isPlayingToAutomotiveHeadUnit const): Deleted.
(WebCore::MediaSessionManagerInterface::supportsSpatialAudioPlaybackForConfiguration): Deleted.
(WebCore::MediaSessionManagerInterface::audioCaptureSourceStateChanged): Deleted.
(WebCore::MediaSessionManagerInterface::audioCaptureSourceCount const): Deleted.
(WebCore::MediaSessionManagerInterface::processIsSuspended const): Deleted.
(WebCore::MediaSessionManagerInterface::isApplicationInBackground const): Deleted.
(WebCore::MediaSessionManagerInterface::isInterrupted const): Deleted.
(WebCore::MediaSessionManagerInterface::addSupportedCommand): Deleted.
(WebCore::MediaSessionManagerInterface::removeSupportedCommand): Deleted.
(WebCore::MediaSessionManagerInterface::supportedCommands const): Deleted.
(WebCore::MediaSessionManagerInterface::scheduleSessionStatusUpdate): Deleted.
(WebCore::MediaSessionManagerInterface::resetSessionState): Deleted.
(WebCore::MediaSessionManagerInterface::supportsSpatialAudioPlayback): Deleted.
(WebCore::MediaSessionManagerInterface::updateSessionState): Deleted.
(WebCore::MediaSessionManagerInterface::pageIdentifier const): Deleted.
* Source/WebCore/platform/audio/PlatformMediaSession.serialization.in:
* Source/WebCore/platform/audio/PlatformMediaSessionInterface.h:
(WebCore::PlatformMediaSessionInterface::mediaSessionGroupIdentifier const):
(WebCore::PlatformMediaSessionInterface::isPlayingOnSecondScreen const):
* Source/WebCore/platform/audio/PlatformMediaSessionManager.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources-output.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Sources.txt:
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.cpp: Added.
(WebKit::RemoteMediaSessionClientProxy):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionClientProxy.h: Added.
(WebKit::RemoteMediaSessionClientProxy::create):
(WebKit::RemoteMediaSessionClientProxy::selectBestMediaSession):
(WebKit::RemoteMediaSessionClientProxy::protectedLogger const):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.cpp: Added.
(WebKit::RemoteMediaSessionManagerProxy::create):
(WebKit::RemoteMediaSessionManagerProxy::RemoteMediaSessionManagerProxy):
(WebKit::RemoteMediaSessionManagerProxy::~RemoteMediaSessionManagerProxy):
(WebKit::RemoteMediaSessionManagerProxy::addSession):
(WebKit::RemoteMediaSessionManagerProxy::removeSession):
(WebKit::RemoteMediaSessionManagerProxy::setCurrentSession):
(WebKit::RemoteMediaSessionManagerProxy::updateSessionState):
(WebKit::RemoteMediaSessionManagerProxy::messageSenderConnection const):
(WebKit::RemoteMediaSessionManagerProxy::messageSenderDestinationID const):
(WebKit::RemoteMediaSessionManagerProxy::sharedPreferencesForWebProcess const):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.h: Added.
(WebKit::RemoteMediaSessionManagerProxy::ref const):
(WebKit::RemoteMediaSessionManagerProxy::deref const):
* Source/WebKit/UIProcess/Media/RemoteMediaSessionManagerProxy.messages.in: Added.
* Source/WebKit/UIProcess/WebPageProxy.cpp:
(WebKit::WebPageProxy::ensureRemoteMediaSessionManagerProxy):
* Source/WebKit/UIProcess/WebPageProxy.h:
* Source/WebKit/UIProcess/WebPageProxy.messages.in:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.cpp: Added.
(WebKit::platformSessionState):
(WebKit::RemoteMediaSessionManager::create):
(WebKit::RemoteMediaSessionManager::RemoteMediaSessionManager):
(WebKit::RemoteMediaSessionManager::~RemoteMediaSessionManager):
(WebKit::RemoteMediaSessionManager::addSession):
(WebKit::RemoteMediaSessionManager::removeSession):
(WebKit::RemoteMediaSessionManager::setCurrentSession):
(WebKit::RemoteMediaSessionManager::updateSessionState):
(WebKit::RemoteMediaSessionManager::messageSenderConnection const):
(WebKit::RemoteMediaSessionManager::messageSenderDestinationID const):
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.h: Added.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionManager.messages.in: Added.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.h: Added.
* Source/WebKit/WebProcess/Media/RemoteMediaSessionState.serialization.in: Copied from Source/WebCore/platform/audio/PlatformMediaSession.serialization.in.
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
(WebKit::m_toolbarsAreVisible):

Canonical link: https://commits.webkit.org/301864@main
